src/transpileCodeblocks/plugin.ts

```diff
@@ -143,19 +148,47 @@ ${lines.slice(Math.max(0, diagnostic.line - 5), diagnostic.line + 6).join('\n')}
   };
 };
 
-function defaultAssembleReplacementNodes(
-  node: CodeNode,
-  file: VFile,
-  virtualFolder: string,
-  virtualFiles: Record<string, VirtualFile>,
-  transpilationResult: Record<string, TranspiledFile>,
-  postProcessTs: PostProcessor,
-  postProcessTranspiledJs: PostProcessor
+function getDefaultAssembleReplacementNodes(
+  compilerSettings: CompilerSettings
 ) {
-  return [
-    {
-      type: 'jsx',
-      value: `
+  return function defaultAssembleReplacementNodes(
```

Comment:
I felt a bit iffy adding this. The purpose was allow access to `compilerSettings` in `defaultAssembleReplacementNodes`, but it didn't feel right to actually pass it as a parameter and change the type signature for `assembleReplacementNodes`